main/mafia.hs

```diff
@@ -502,6 +509,17 @@ mafiaScript :: File -> [Argument] -> EitherT MafiaError IO ()
 mafiaScript file args =
   firstT MafiaScriptError $ runScript file args
 
+mafiaExec :: [Argument] -> EitherT MafiaError IO ()
+mafiaExec args =
+  liftCabal $ cabal_ "exec" $ fixupArgs args
+  where
+    fixupArgs = \case
+      [] -> [] -- Should not happen.
+      [x] -> [x] -- A command without arguments.
+      -- Insert `--` between the command and its arguments so cabal doesn't
+      -- mess with them.
+      (x:xs) -> x : "--" : xs
```

Comment:
> why is that?

I think it makes a function easier to read if you can follow it from top to bottom rather jumping around to follow the flow of data.

It is also more consistent with many of our other projects. When I first started at Ambiata I used `where` a lot, but after reading the source of a bunch of our projects I found that it was not used at all and the resulting code was easier to follow as a result, so I decided to stop using it myself in the name of consistency.